src/clj/puppetlabs/http/client/async.clj

```diff
@@ -175,12 +176,13 @@
   ([opts :- common/RawUserRequestOptions
     callback :- common/ResponseCallbackFn
     client :- HttpAsyncClient]
-    (request-with-client opts callback client nil nil))
+    (request-with-client opts callback client nil nil true))
   ([opts :- common/RawUserRequestOptions
     callback :- common/ResponseCallbackFn
     client :- HttpAsyncClient
     metric-registry :- (schema/maybe MetricRegistry)
-    metric-namespace :- (schema/maybe schema/Str)]
+    metric-namespace :- (schema/maybe schema/Str)
+    use-url-metrics :- (schema/maybe schema/Bool)]
```

Comment:
I suppose it we wanted to make this fully backward-compatible, we should introduce a third arity for the `request-with-client` function which takes the new `enable-url-metrics?` parameter - so that prior clients using the arity that has `metric-registry` and `metrics-namespace` would be unaffected.